app/models/ability.rb

```diff
@@ -0,0 +1,48 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    user || User.new
+    # Rules
+    ###if user.admin?
+    ###  can :manage, :all
+    ###elsif user == current_user
+    ###  can :manage, :all
+    ###else
+    ###  can :read, :all
+    ###end
+    if user.admin?
+      can :manage, :all
+    end
+    can :manage, Post do |p|
+      p.user == user
+    end
+
```

Comment:
Maybe you have it implemented somewhere else, but would you want a cancan method for whether or not users can favourite/unfavorite a post?